### DISCUSSION TITLE
Building OM With Homebrew Installed

### DISCUSSION THREAD

**Discussion by ptoshkov** [OpenModelica/OpenModelica#9993] [Show and tell]
Dear all,

I wanted to build OpenModelica on Macintosh. The guide suggests using Macports:
https://github.com/OpenModelica/OpenModelica/blob/master/README.cmake.md#33-macos

I already have the package manager Homebrew installed. I didn’t want to install Macports, as they can be in conflict sometimes. To add fuel to the fire, I couldn’t find all of OM’s dependencies on Homebrew (namely, qt5-qtwebkit).

The solution for me was to install Macports for the current user only. This is how:

To install Macports, obtain the latest tarball from https://distfiles.macports.org/MacPorts/MacPorts-2.8.0.tar.bz2 and extract it
```sh
$ cd MacPorts-2.8.0/
$ ./configure --enable-readline --prefix=$HOME/macports --with-no-root-privileges --with-install-user=$USER
$ make
$ make install
```

To install dependencies (this takes a few hours…)
```sh
$ cd ~/macports/bin
$ ./port selfupdate
$ ./port install cmake ccache qt5 qt5-qtwebkit autoconf boost OpenSceneGraph openjdk11
```

To compile OM...

i) Make sure Macports QT can be found
```sh
$ export PATH="$HOME/macports/libexec/qt5/bin:$PATH"
$ export LDFLAGS="-L$HOME/macports/libexec/qt5/lib"
$ export CPPFLAGS="-I$HOME/macports/libexec/qt5/include"
$ export PKG_CONFIG_PATH="$HOME/macports/libexec/qt5/lib/pkgconfig“
```

ii) specify $HOME/macports as the CMake prefix
```sh
$ cmake -S . -B build_cmake -DCMAKE_PREFIX_PATH=$HOME/macports -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_Fortran_COMPILER=gfortran
$ cmake --build build_cmake --parallel <Nr. of cores> --target install
```

I hope this is useful.

**Comment by mahge** [comment]
Thank you for the investigation and detailed instructions. I will try to add it to the user guide.

--------------------------------------
Out of curiosity, 
> i) Make sure Macports QT can be found
> 
> ```shell
> $ export PATH="$HOME/macports/libexec/qt5/bin:$PATH"
> $ export LDFLAGS="-L$HOME/macports/libexec/qt5/lib"
> $ export CPPFLAGS="-I$HOME/macports/libexec/qt5/include"
> $ export PKG_CONFIG_PATH="$HOME/macports/libexec/qt5/lib/pkgconfig“
> ```

Are you sure that this is needed? Does it no suffice to just specify `-DCMAKE_PREFIX_PATH=$HOME/macports`. If it is enough then we can simplify these instructions a bit.

**Comment by ptoshkov** [comment]
Yes, this is enough:
`-DCMAKE_PREFIX_PATH=$HOME/macports
`

You don't necessarily need
`$ export PATH="$HOME/macports/libexec/qt5/bin:$PATH"`
`$ export LDFLAGS="-L$HOME/macports/libexec/qt5/lib"`
`$ export CPPFLAGS="-I$HOME/macports/libexec/qt5/include"`
`$ export PKG_CONFIG_PATH="$HOME/macports/libexec/qt5/lib/pkgconfig“`
